include/nccl_ofi.h

```diff
@@ -5,7 +5,8 @@
 #ifndef NCCL_OFI_H_
 #define NCCL_OFI_H_
 
-#ifdef _cplusplus
```

Comment:
There's been a lot of copy-pasta in this code base.  I don't actually think we should force C linkage for most of these headers.  `include/nccl_api.h` and all the `include/nccl-headers` files obviously need C linkage.  I'm not positive about nccl_log.h; need to audit that a bit closer.  But this file and the data structure headers shouldn't need C linkage (and having C linkage is going to make it difficult to C++-ify later).

Reply:
overall I agree; longterm I would like to see that we do not require C linkage for anything that isn't exported and I'd like to take advantage of inlining and disabling interposition, but the goal of this PR is just to get the unit tests compiling under C++ as a baby step, not to oneshot port the codebase broadly to C++, and I'd rather tackle this later when we have time to reason about it.

These guards already existed and I'm not introducing anything new here, they were just broken because they had the wrong token.

Reply:
What I would like to do eventually:

```
src/include/ <--- internal usage only, do whatever you want in here. Not installed.
include/     <---  installed alongside the plugin, and includes:
                   * anything that needs to be visible to NCCL
                   * any extensibility API we want to define (such as the tweaks we currently expose 
                     for platform-aws) that are used to customize the plugin.
```

Reply:
i'd ditch the src/include and just put it all in src/

Reply:
I'd like to have it structured by namespace, so there's good reason to have a second include directory, but we can cross that bridge when we get there.